### ISSUE TITLE
[Question] Why discard the last element of logit.

### ISSUE THREAD

**Issue by ChenMnZ** [Cornell-RelaxML/quip-sharp#48]
Hi, 

I curious about the reason of discarding last element of prediction logit during training (corresponding to `[:, :-1]`). As shown in the follows.
https://github.com/Cornell-RelaxML/quip-sharp/blob/b22288cbb71abf997ea15c9858da13f4de3cc005/quantize_llama/finetune_e2e_llama.py#L102

**Comment by tsengalb99** [comment]
IIRC this is b/c the logits outputted by huggingface are "shifted by one" so the last logit (-1) is a new prediction.

**Comment by ChenMnZ** [comment]
Thanks for answering.
